hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsHttpOperation.java

```diff
@@ -336,95 +347,141 @@ public void sendRequest(byte[] buffer, int offset, int length) throws IOExceptio
    *
    * @throws IOException if an error occurs.
    */
-  public void processResponse(final byte[] buffer, final int offset, final int length) throws IOException {
+  public void processResponse(final byte[] buffer,
+      final int offset,
+      final int length) throws IOException {
 
     // get the response
     long startTime = 0;
-    if (this.isTraceEnabled) {
+    if (isTraceEnabled) {
       startTime = System.nanoTime();
     }
 
-    this.statusCode = this.connection.getResponseCode();
+    statusCode = connection.getResponseCode();
 
-    if (this.isTraceEnabled) {
-      this.recvResponseTimeMs = elapsedTimeMs(startTime);
+    if (isTraceEnabled) {
+      recvResponseTimeMs = elapsedTimeMs(startTime);
     }
 
-    this.statusDescription = this.connection.getResponseMessage();
+    statusDescription = connection.getResponseMessage();
 
-    this.requestId = this.connection.getHeaderField(HttpHeaderConfigurations.X_MS_REQUEST_ID);
-    if (this.requestId == null) {
-      this.requestId = AbfsHttpConstants.EMPTY_STRING;
+    requestId = connection.getHeaderField(HttpHeaderConfigurations.X_MS_REQUEST_ID);
+    if (requestId == null) {
+      requestId = AbfsHttpConstants.EMPTY_STRING;
     }
     // dump the headers
     AbfsIoUtils.dumpHeadersToDebugLog("Response Headers",
         connection.getHeaderFields());
 
-    if (AbfsHttpConstants.HTTP_METHOD_HEAD.equals(this.method)) {
+    if (AbfsHttpConstants.HTTP_METHOD_HEAD.equals(method)) {
       // If it is HEAD, and it is ERROR
       return;
     }
 
-    if (this.isTraceEnabled) {
+    if (isTraceEnabled) {
       startTime = System.nanoTime();
     }
 
+    long totalBytesRead = 0;
```

Comment:
nit: totalBytesRead doesn't look needed here. 

This logic can be simplified to:
1. Set bytesReceived initially to 0
2. Assign bytesReceived in the try statement
3. Remove the assignment of bytesReceived in the finally statement